manila/share/drivers/netapp/dataontap/cluster_mode/lib_base.py

```diff
@@ -2178,20 +2196,25 @@ def _manage_container(self, share, vserver, vserver_client):
                 raise exception.ManageExistingShareTypeMismatch(
                     reason=msg % msg_args)
 
-        share_name = self._get_backend_share_name(share['id'])
-        debug_args = {
-            'share': share_name,
-            'aggr': (",".join(aggregate_name) if flexgroup_vol
-                     else aggregate_name),
-            'options': provisioning_options
-        }
-        LOG.debug('Managing share %(share)s on aggregate(s) %(aggr)s with '
-                  'provisioning options %(options)s', debug_args)
+        rename = self._get_driver_option(options, 'rename', False, bool)
+        if rename:
+            share_name = self._get_backend_share_name(share['id'])
+            debug_args = {
+                'share': share_name,
+                'aggr': (",".join(aggregate_name) if flexgroup_vol
+                         else aggregate_name),
+                'options': provisioning_options
+            }
+            LOG.debug('Managing share %(share)s on aggregate(s) %(aggr)s with '
+                      'provisioning options %(options)s', debug_args)
+        else:
+            share_name = volume_name
 
         # Rename & remount volume on new path.
-        vserver_client.unmount_volume(volume_name)
-        vserver_client.set_volume_name(volume_name, share_name)
-        vserver_client.mount_volume(share_name)
+        if volume_name != share_name:
```

Comment:
why are these behind a separate condition and not in the if rename: block above?